test/metabase/driver/sql_jdbc/actions_test.clj

```diff
@@ -51,3 +53,139 @@
                                             (actions/perform-action! :row/delete (mt/mbql-query categories {:filter [:= $id 58]})))))
           (testing "Make sure our impl was actually called."
             (is @parse-sql-error-called?)))))))
+
+(mt/defdataset action-error-handling
+  [["group"
+    [{:field-name "name" :base-type :type/Text :not-null? true}
+     {:field-name "ranking" :base-type :type/Integer :not-null? true :unique? true}]
+    [["admin" 1]
+     ["user" 2]]]
+   ["user"
+    [{:field-name "name" :base-type :type/Text :not-null? true}
+     {:field-name "group-id" :base-type :type/Integer :fk "group" :not-null? true}]
+    [["crowberto" 1]
+     ["rasta"     2]
+     ["lucky"     1]]]])
+
+(defn- perform-action-ex-data
+  [& args]
+  (try
+   (apply actions/perform-action! args)
+   (catch Exception e
+     (ex-data e))))
+
+(deftest action-error-handling-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :actions)
```

Comment:
I think this should be restricted to JDBC drivers for the future implementors of actions for non-jdbc drivers to skip these tests
```suggestion
  (mt/test-drivers (filter #(isa? driver/hierarchy % :sql-jdbc) (mt/normal-drivers-with-feature :actions))
```